damus/Views/Relays/RelayDetailView.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  RelayDetailView.swift
+//  damus
+//
+//  Created by Joel Klabo on 2/1/23.
+//
+
+import SwiftUI
+
+struct RelayDetailView: View {
+    let relay: String
+    
+    @State private var errorString: String?
+    @State private var nip11: RelayNIP11?
+    
+    @State var conn_color: Color
+        
+    var body: some View {
+        Group {
+            if let nip11 {
+                Form {
+                    Section(NSLocalizedString("Name", comment: "Label to display relay name.")) {
+                        Text(nip11.name)
+                    }
+                    Section(NSLocalizedString("Relay", comment: "Label to display relay address.")) {
+                        HStack {
+                            Text(relay)
+                            Spacer()
+                            Circle()
+                                .frame(width: 8.0, height: 8.0)
+                                .foregroundColor(conn_color)
+                        }
+                    }
+                    Section(NSLocalizedString("Description", comment: "Label to display relay description.")) {
+                        Text(nip11.description)
+                    }
+                    Section(NSLocalizedString("Public Key", comment: "Label to display relay contact public key.")) {
+                        Text(nip11.pubkey)
+                    }
+                    Section(NSLocalizedString("Contact", comment: "Label to display relay contact information.")) {
+                        Text(nip11.contact)
+                    }
+                    Section(NSLocalizedString("Software", comment: "Label to display relay software.")) {
+                        Text(nip11.software)
+                    }
+                    Section(NSLocalizedString("Version", comment: "Label to display relay software version.")) {
+                        Text(nip11.version)
+                    }
+                    Section(NSLocalizedString("Supported NIPs", comment: "Label to display relay's supported NIPs.")) {
+                        Text(nipsList(nips: nip11.supported_nips))
+                    }
+                }
+            } else if let errorString {
+                Text(errorString)
+                    .foregroundColor(.red)
+                    .font(.caption)
+            } else {
+                ProgressView()
+            }
+        }
+        .navigationTitle(nip11?.name ?? "")
+        .onAppear {
```

Comment:
you can do `.task` instead of `onAppear` and then use the async version of URLSession.shared...

Reply:
I'll make this change don't worry about it

Reply:
Oh nice, didn't know that. Thanks!